qiskit/circuit/library/standard_gates/x.py

```diff
@@ -851,9 +851,19 @@ def __new__(cls, num_ctrl_qubits=None, label=None, ctrl_state=None):
             return gate
         return super().__new__(cls)
 
-    def __init__(self, num_ctrl_qubits, label=None, ctrl_state=None, _name='mcx'):
+    def __init__(self, num_ctrl_qubits, label=None, ctrl_state=None, _name=None):
         """Create new MCX gate."""
         num_ancilla_qubits = self.__class__.get_num_ancilla_qubits(num_ctrl_qubits)
+
+        # if MCXGate is not specified via MCXGrayCode, MCXRecursive, MCXVChain
+        if _name is None:
```

Comment:
Should we capture the corner case of `num_ctrl_qubits=0`? 🙂 Or is that covered by the `ControlledGate` superclass, i.e. does it raise an error in `super().init()` below if `num_ctrl_qubits` is 0?

Reply:
I think it's in `qiskit/circuit/controlledgate.py`:
```
CircuitError: 'The number of control qubits must be in [1, num_qubits-1]'
```